### PR TITLE
Update kubeadm supported etcd version to 3.2.14 in 1.10

### LINKED ISSUE #621: Update default etcd version to 3.2.x in v1.10
Same as https://github.com/kubernetes/kubeadm/issues/450, but for 3.2.12 or something for v1.10

@xiangpengzhao is on it already.
We're following the upstream k8s validated version, relevant PRs/issues: https://github.com/kubernetes/kubernetes/pull/57230, https://github.com/kubernetes/kubernetes/pull/57480,  https://github.com/kubernetes/kubernetes/issues/56114 and https://github.com/kubernetes/kubernetes/issues/47131

### PR DESCRIPTION
**Pull request by xiangpengzhao**
**What this PR does / why we need it**:
Kubernetes will upgrade to etcd server 3.2.14 in 1.10 cycle (#58645), update DefaultEtcdVersion in kubeadm to this version.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
relevant PR: #57480 #58645
fixes: https://github.com/kubernetes/kubeadm/issues/621

**Special notes for your reviewer**:
/cc @kubernetes/sig-cluster-lifecycle-pr-reviews

**Release note**:

```release-note
NONE
```
kubeadm don't need to advertise this in release notes.

### CODE REVIEW

**Comment by xiangpengzhao** [comment]
/hold
until it's ready.

File: cmd/kubeadm/app/constants/constants.go
Comment (luxas) [inline cmd/kubeadm/app/constants/constants.go]: you can remove this one now

File: cmd/kubeadm/app/constants/constants.go
Comment (luxas) [inline cmd/kubeadm/app/constants/constants.go]: this _might_ be bumped to 3.2.11 or something higher as well before things are merged in core in other places

**Review by luxas — APPROVED** [review]
Thanks for the quick patch!

Basically LGTM, but we should wait for https://github.com/kubernetes/kubernetes/pull/57160 to merge first.
cc @jpbetz @wojtek-t @kubernetes/sig-cluster-lifecycle-pr-reviews 

Please fix the nits; meanwhile I'll put a hold on this one
/hold

File: cmd/kubeadm/app/constants/constants.go
Comment (xiangpengzhao) [reply cmd/kubeadm/app/constants/constants.go]: :+1: updated.

File: cmd/kubeadm/app/constants/constants.go
Comment (xiangpengzhao) [reply cmd/kubeadm/app/constants/constants.go]: yeah let's hold until it's decided.

**Comment by luxas** [comment]
@xiangpengzhao Need to fix the unit tests

**Comment by xiangpengzhao** [comment]
unit tests fixed. Now pending https://github.com/kubernetes/kubernetes/pull/57480

File: cmd/kubeadm/app/constants/constants.go
Comment (xiangpengzhao) [inline cmd/kubeadm/app/constants/constants.go]: @luxas MinExternalEtcdVersion should be the supported etcd version of 1.9, right?

File: cmd/kubeadm/app/constants/constants.go
Comment (xiangpengzhao) [inline cmd/kubeadm/app/constants/constants.go]: yeah it's too early to add supported etcd version of 1.11. Add it here to make the test of "experimental version" pass. It may change in the future.

Comment (luxas) [reply cmd/kubeadm/app/constants/constants.go]: looks good to me

File: cmd/kubeadm/app/constants/constants.go
Comment (luxas) [reply cmd/kubeadm/app/constants/constants.go]: cool

**Review by luxas — APPROVED** [review]
/lgtm

Thanks @xiangpengzhao!

**Comment by xiangpengzhao** [comment]
rebased after https://github.com/kubernetes/kubernetes/pull/57271 merged.

**Review by luxas — APPROVED** [review]
/lgtm

**Comment by timothysc** [comment]
Yeah, let's hold on this until we get confirmation that is passed the scale tests for a cycle. 

/cc @wojtek-t

**Comment by xiangpengzhao** [comment]
Pending #57480 and the target etcd version is 3.2.13 now. Will change it after that PR gets merged.

**Comment by timothysc** [comment]
I'd still wait for some timeout, there has been some reverts.

**Comment by xiangpengzhao** [comment]
@timothysc agreed. It'd be better waiting until it's proved to be stable :)

**Review by timothysc — APPROVED** [review]
/lgtm 
/approve

**Comment by errordeveloper** [comment]
@xiangpengzhao please rebase.

**Comment by timothysc** [comment]
Please update to 3.2.14 per other issue referenced.

**Comment by timothysc** [comment]
@xiangpengzhao ping re: update, I'll timeout by next week, but we need to get this change in.

File: cmd/kubeadm/app/constants/constants.go
Comment (xiangpengzhao) [inline cmd/kubeadm/app/constants/constants.go]: We have changed etcd version to 3.1.11 for 1.9 (https://github.com/kubernetes/kubernetes/pull/57811). We should change the version here as well, right? But if so, it may cause issue when starting a 1.9.0 against HEAD kubeadm? @luxas @timothysc

**Comment by xiangpengzhao** [comment]
@timothysc rebased and updated. But seems that etcd client hasn't been updated yet in the ref issues, so we may still need to hold for a while.

Another concern posted in https://github.com/kubernetes/kubernetes/pull/57230#discussion_r165584519

File: cmd/kubeadm/app/constants/constants_test.go
Comment (timothysc) [inline cmd/kubeadm/app/constants/constants_test.go]: .99?

File: cmd/kubeadm/app/constants/constants.go
Comment (timothysc) [reply cmd/kubeadm/app/constants/constants.go]: Yes.

**Review by timothysc — COMMENTED** [review]
Minor stuff then lgtm

/approve

**Review by timothysc — APPROVED** [review]
/lgtm 
/approve

**Comment by xiangpengzhao** [comment]
xref: https://github.com/kubernetes/kubernetes/pull/58645#issuecomment-362652010

**Comment by xiangpengzhao** [comment]
`pull-kubernetes-bazel-test` fails repeatedly. I don't find any clue with the change :(

**Pull request merged**
